spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/ValidationMethod.scala

```diff
@@ -219,7 +219,325 @@ class Top1Accuracy[T: ClassTag](
 }
 
 /**
- * Caculate the percentage that target in output's top5 probability indexes
+ * Calculate the Mean Average Precision (MAP). The algorithm follows VOC Challenge after 2010
+ */
+class MeanAveragePrecision[T: ClassTag](k: Int, classes: Int)(
+  implicit ev: TensorNumeric[T]) extends ValidationMethod[T] {
+
+  require(classes > 0 && classes <= classes, s"The number of classes should be "
+    + s"> 0 and <= $classes, but got $classes")
+  require(k > 0, s"k should be > 0, but got $k")
```

Comment:
what does K mean

Reply:
K is the number of predictions (top-K confident items) that are calculated in AP. It is used in MAP of classification.  For object detection, it is set to -1 (count for all prediction)

Reply:
so please add class parameter comments

Reply:
comments updated